lib/calendar.rb

```diff
@@ -0,0 +1,40 @@
+require 'date'
+
+module Hotel
+  
+  class Calendar
+    
+    attr_reader :today
+    
+    def initialize
+      @today = Date.today
+    end
+    
+    def valid_start_date?(start_date)
+      # Does not accept reservations made on the current day
+      start_date > @today ? true : false
+    end
+    
+    def valid_end_date?(start_date, end_date)
+      # Does not accept one-day reservations
+      start_date < end_date ? true : false
+    end
+    
+    def overlap?(dates_to_check, reserved_dates)
+      reserved_dates.flatten!
+      (reserved_dates & dates_to_check).any? # if true, then there is overlap.
+    end
+    
+    def get_date_range(start_date, end_date)
+      if !valid_start_date?(start_date)
```

Comment:
Clever, but was this a requirement for our project? :)

Also, consider this: What happens to your tests in a year?

Otherwise, I really like how this method/class is structured: it separately has methods to determine if one property is valid. Keep this pattern up!